src/browser-polyfill.js

```diff
@@ -6,11 +6,11 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 "use strict";
 
-if (typeof globalThis != "object" || typeof chrome != "object" || !chrome || !chrome.runtime || !chrome.runtime.id) {
+if (!globalThis.chrome?.runtime?.id) {
   throw new Error("This script should only be loaded in a browser extension.");
 }
 
-if (typeof globalThis.browser === "undefined" || Object.getPrototypeOf(globalThis.browser) !== Object.prototype) {
+if (!globalThis.browser?.runtime?.id || Object.getPrototypeOf(globalThis.browser) !== Object.prototype) {
```

Comment:
Note: this code should be further refined due to:

- https://github.com/mozilla/webextension-polyfill/issues/364